internal/server/singleprocess/service_job.go

```diff
@@ -337,7 +337,7 @@ func (s *service) onDemandRunnerStartJob(
 
 	// Arguments for the runner image. Waypoint is ALWAYS assumed to be
 	// the entrypoint for ODR images.
-	args := []string{"runner", "agent", "-vv", "-id", runnerId, "-odr"}
+	args := []string{"runner", "agent", "-vv", "-id", runnerId, "-odr", "-odr-profile-id", od.Id}
```

Comment:
I don't love this way of setting metadata on the runner - it feels very indirect. It would feel much cleaner to me if we called a server api here-ish to create a new "pending" `runner` record of type odr, then just start the runner with the predetermined ID. Then, when the runner registered itself to the server, it would just set its state from "pending" to "active", or something.

It also feels a little weird from a security perspective - fresh runners have the opportunity to lie about their profile id and odr-ness if we rely on them to tell the server. 

If and when it's time to add the next runner flag like this, I think we should revisit this pattern.

Reply:
We need to add some sort of runner "adoption" or "activation" step for some future work we have planned for exactly the reason you stated. Ideally: runners can relatively easy ATTEMPT to register with the server, but a human (or machine via more privileged API) has to "adopt" the runner to activate it. I think we can do this using some sort of certificate exchange mechanism (so if the runner restarts or something it can prove its adoption for a period of time). I have various thoughts here.

For now, I think this is okay.